boards/DCM/Src/App/states/App_InitState.c

```diff
@@ -24,11 +25,83 @@ static void InitStateRunOnTick100Hz(struct StateMachine *const state_machine)
 
     struct DcmWorld *world = App_SharedStateMachine_GetWorld(state_machine);
     struct DcmCanTxInterface *can_tx_interface = App_DcmWorld_GetCanTx(world);
+    struct DcmCanRxInterface *can_rx_interface = App_DcmWorld_GetCanRx(world);
+    struct ErrorTable *       error_table = App_DcmWorld_GetErrorTable(world);
+    struct InverterSwitches * inverter_switches =
+        App_DcmWorld_GetInverterSwitches(world);
 
     App_CanTx_SetPeriodicSignal_TORQUE_REQUEST(can_tx_interface, 0.0f);
+    // Holds previous start switch position (true = UP/ON, false = DOWN/OFF)
+    static bool prev_start_switch_position =
+        true; // Initialize to true to prevent a false start
 
-    // No need for any safety checks, just run! (this is a demo)
+#ifdef DEBUG // Enter drive state directly from init state
     App_SharedStateMachine_SetNextState(state_machine, App_GetDriveState());
+    return;
+#endif
+
+    const bool both_airs_closed =
+        App_CanRx_BMS_AIR_STATES_GetSignal_AIR_POSITIVE(can_rx_interface) ==
+            CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_CLOSED_CHOICE &&
+        App_CanRx_BMS_AIR_STATES_GetSignal_AIR_NEGATIVE(can_rx_interface) ==
+            CANMSGS_BMS_AIR_STATES_AIR_NEGATIVE_CLOSED_CHOICE;
+
+    const bool inverter_faulted =
+        App_CanRx_INVL_INTERNAL_STATES_GetSignal_D1_VSM_STATE_INVL(
+            can_rx_interface) ==
+            CANMSGS_INVL_INTERNAL_STATES_D1_VSM_STATE_INVL_BLINK_FAULT_CODE_STATE_CHOICE ||
+        App_CanRx_INVR_INTERNAL_STATES_GetSignal_D1_VSM_STATE_INVR(
+            can_rx_interface) ==
+            CANMSGS_INVR_INTERNAL_STATES_D1_VSM_STATE_INVR_BLINK_FAULT_CODE_STATE_CHOICE;
+
+    // Provide LV to inverters when both AIRS are closed and DC bus voltage ~
+    // 400 V
+    if (both_airs_closed && !App_InverterSwitches_IsRightOn(inverter_switches))
+    {
+        App_InverterSwitches_TurnOnRight(inverter_switches);
+    }
+    if (both_airs_closed && !App_InverterSwitches_IsLeftOn(inverter_switches))
+    {
+        App_InverterSwitches_TurnOnLeft(inverter_switches);
+    }
+
+    // Transition to fault state if an inverter has faulted or
+    // a critical error has been set
+    if (inverter_faulted ||
```

Comment:
might be easier to read if do something like this

https://github.com/UBCFormulaElectric/Consolidated-Firmware/pull/730/files#r812667678

Reply:
lmk if the new state transition code is easier to follow